### ISSUE TITLE
Grid items are there but not visible

### ISSUE THREAD

**Issue by CD1010** [tiberiuzuld/angular-gridster2#125]
I tried making a simple sample from readme.
I made and angular 4 component that has the following markup:
<div>
    <gridster [options]="options">
        
            <gridster-item [item]="item" *ngFor="let item of gridItems" >
                <div class="gridster-item-content">
                    Some content to click without dragging the widget
                </div>
            </gridster-item>
</gridster>
</div>

and ts file has this:

import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { GridsterItem } from 'angular-gridster2';
import { GridsterConfig } from 'angular-gridster2';

@Component({
    selector: 'dashboard',
    templateUrl: './dashboard.component.html',
    styleUrls: ['./dashboard.component.css'],
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class DashboardComponent implements OnInit {
    options: GridsterConfig;
    gridItems: Array<GridsterItem>;

    static eventStop(item : GridsterItem, itemComponent: GridsterItem, event: any[]) {
        console.info('eventStop', item, itemComponent, event);
    }

    static itemChange(item: GridsterItem, itemComponent: GridsterItem) {
        console.info('itemChanged', item, itemComponent);
    }

    static itemResize(item: GridsterItem, itemComponent: GridsterItem) {
        console.info('itemResized', item, itemComponent);
    }

    static itemInit(item: GridsterItem, itemComponent: GridsterItem) {
        console.info('itemInitialized', item, itemComponent);
    }

    emptyCellClick(event: any[], item: GridsterItem) {
        console.info('empty cell click', event, item);
        this.gridItems.push(item);
    }

    ngOnInit() {
        this.options = {
            gridType: 'fit',
            compactType: 'none',
            itemChangeCallback: DashboardComponent.itemChange,
            itemResizeCallback: DashboardComponent.itemResize,
            itemInitCallback: DashboardComponent.itemInit,
            margin: 5,
            outerMargin: true,
            mobileBreakpoint: 640,
            minCols: 1,
            maxCols: 100,
            minRows: 1,
            maxRows: 100,
            maxItemCols: 100,
            minItemCols: 1,
            maxItemRows: 100,
            minItemRows: 1,
            maxItemArea: 2500,
            minItemArea: 1,
            defaultItemCols: 1,
            defaultItemRows: 1,
            fixedColWidth: 105,
            fixedRowHeight: 105,
            keepFixedHeightInMobile: false,
            keepFixedWidthInMobile: false,
            scrollSensitivity: 10,
            scrollSpeed: 20,
            enableEmptyCellClick: false,
            enableEmptyCellContextMenu: false,
            enableEmptyCellDrop: false,
            enableEmptyCellDrag: false,
            emptyCellClickCallback: this.emptyCellClick.bind(this),
            emptyCellContextMenuCallback: this.emptyCellClick.bind(this),
            emptyCellDropCallback: this.emptyCellClick.bind(this),
            emptyCellDragCallback: this.emptyCellClick.bind(this),
            emptyCellDragMaxCols: 50,
            emptyCellDragMaxRows: 50,
            draggable: {
                enabled: true,
                ignoreContentClass: 'gridster-item-content',
                ignoreContent: false,
                dragHandleClass: 'drag-handler',
                stop: DashboardComponent.eventStop
            },
            resizable: {
                enabled: true,
                stop: DashboardComponent.eventStop,
                handles: {
                    s: true,
                    e: true,
                    n: true,
                    w: true,
                    se: true,
                    ne: true,
                    sw: true,
                    nw: true
                }
            },
            api: {
                resize: DashboardComponent.eventStop,
                optionsChanged: DashboardComponent.eventStop,
                getNextPossiblePosition: DashboardComponent.eventStop,
            },
            swap: false,
            pushItems: true,
            pushResizeItems: false,
            displayGrid: 'onDrag&Resize',
            disableWindowResize: false
        };

        this.gridItems = [
            { cols: 2, rows: 1, y: 0, x: 0 },
            { cols: 2, rows: 2, y: 0, x: 2, hasContent: true },
            { cols: 1, rows: 1, y: 0, x: 4 },
            { cols: 1, rows: 1, y: 2, x: 5 },
            { cols: 2, rows: 2, y: 3, x: 5, minItemRows: 2, minItemCols: 2, label: 'Min rows & cols = 2' },
            { cols: 2, rows: 2, y: 2, x: 0, maxItemRows: 2, maxItemCols: 2, label: 'Max rows & cols = 2' },
            { cols: 2, rows: 1, y: 2, x: 2, dragEnabled: true, resizeEnabled: true, label: 'Drag&Resize Enabled' },
            { cols: 1, rows: 1, y: 2, x: 4, dragEnabled: false, resizeEnabled: false, label: 'Drag&Resize Disabled' },
            { cols: 1, rows: 1, y: 2, x: 6, initCallback: DashboardComponent.itemInit }
        ];
    }

    changedOptions() {
        if (this.options.api && this.options.api.optionsChanged) {
            this.options.api.optionsChanged();
        }
    }

    removeItem($event: any, item: any) {
        $event.preventDefault();
        $event.stopPropagation();
        this.gridItems.splice(this.gridItems.indexOf(item), 1);
    }

    addItem() {
        this.gridItems.push({});
    }
}

what could be wrong?
When i inspect in browser, i see all the gridster items getting created in dom, but screen is blank.

**Comment by celloman** [comment]
Try giving your grid a height, I believe with gridType: 'fit', the height of the rows will be the height of the full grid/number of rows - margins. If your grid has a height of 0, nothing will be visible.

I was able to bind to [style.height.px] on gridster in the template and perform the following operation (called from my item init and item change callbacks). This keeps the grid height matched up with the number of rows. Note, this assumes a grid of squares with no outer margins. I saved the gridster reference away within the init callback.

```
calculateGridHeight() {
    if (this.gridster) {
      this.gridster.calculateLayout();
      this.gridHeight = this.gridster.curColWidth * this.gridster.rows - this.gridster.options.margin;
    }
}
